From 3aa1263dae09f6ad0d085e711d32134816d33d9b Mon Sep 17 00:00:00 2001
From: Ruslan Bay <67730802+ruslanbay@users.noreply.github.com>
Date: Tue, 17 Feb 2026 12:47:09 +0000
Subject: [PATCH 55/56] Count video device entities to set nr_queues: During
 graph walk in ipu_isys_video_prepare_streaming(), count all video device
 entities and increment nr_queues accordingly. The streaming start logic waits
 for nr_streaming == nr_queues, but nr_queues was only incremented in
 isa_link_validate(). For direct CSI-2 capture pipelines (non-ISA), nr_queues
 stayed at 0 causing the stream start to hang indefinitely.

---
 drivers/media/pci/intel/ipu-isys-video.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-isys-video.c b/drivers/media/pci/intel/ipu-isys-video.c
index 9906ec66330e..f7bbfbc97e21 100644
--- a/drivers/media/pci/intel/ipu-isys-video.c
+++ b/drivers/media/pci/intel/ipu-isys-video.c
@@ -1936,11 +1936,14 @@ int ipu_isys_video_prepare_streaming(struct ipu_isys_video *av,
 	if (rval)
 		goto out_pipeline_stop;
 
-	/* Gather all entities in the graph. */
+	/* Gather all entities in the graph and count video queues. */
 	mutex_lock(&mdev->graph_mutex);
 	media_graph_walk_start(&graph, &av->vdev.entity);
-	while ((entity = media_graph_walk_next(&graph)))
+	while ((entity = media_graph_walk_next(&graph))) {
 		media_entity_enum_set(&ip->entity_enum, entity);
+		if (is_media_entity_v4l2_video_device(entity))
+			ip->nr_queues++;
+	}
 
 	mutex_unlock(&mdev->graph_mutex);
 
-- 
2.51.0

